src/gt4py/next/iterator/transforms/symbol_ref_utils.py

```diff
@@ -13,44 +13,73 @@
 # SPDX-License-Identifier: GPL-3.0-or-later
 
 import dataclasses
-from typing import Iterable, Sequence
+from typing import Iterable, Optional, Sequence
 
 import gt4py.eve as eve
 from gt4py.next.iterator import ir as itir
 
 
 @dataclasses.dataclass
 class CountSymbolRefs(eve.NodeVisitor):
-    ref_counts: dict[str, int]
+    ref_counts: dict[str, int] = dataclasses.field(default_factory=dict)
```

Comment:
```suggestion
    ref_counts: dict[str, int] = dataclasses.field(default_factory=lambda: defaultdict(int))
```
does this work to simplify line 62-64?

Reply:
Didn't know that one. Yes, it does.